### DISCUSSION TITLE
Keycloak and IIS web application

### DISCUSSION THREAD

**Discussion by abahmanem** [keycloak/keycloak#35472] [Q&A]
Hi

I have keycloak installed in our DMZ and is accessible from outside. 

We have 2 web applications hosted internally in IIS.

I want to add them in keycloak to be accessible with SSO.

Can we use Windows Integrated authentication  in keycloak ?

I want also to proxy those applications : 

URL of keycloak : 

https://idp.iam.ch

Internal URL of IIS web application 

https://iam.intern-web.app.ch/app

This URL is not accessible from outside 

I want to access this app from dmz with SSO using keycloak with this URL : 

https://idp.iam.ch/app

So does keycloak play role of a proxy or not?

Thank you

**Comment by codespearhead** [comment]
> Does Keycloak act as a proxy or not?

If the framework used in the backend of your applications includes an OIDC/OAuth2 client, like `spring-boot-starter-oauth2-client` in Spring Boot, those applications can interact directly with Keycloak. In that case, there's no need for an API gateway between Keycloak and your applications. However, if such a client is not available, you'll need an API gateway to ensure that only requests with valid tokens and the required scopes issued by Keycloak are allowed to reach your applications.

Maintaining OIDC/OAuth2 clients is the responsibility of your framework, not the Keycloak project.

See [01], [02], [03], and [04] for more information.

Does that answer your question? If so, please mark this comment as the accepted answer [05] for this discussion, so it no longer appears as unanswered in the GitHub discussions tab.

***

[01] https://www.krakend.io/features#:~:text=OAuth2
[02] https://tyk.io/docs/api-management/client-authentication/#using-the-authorization-code-grant
[03] https://developers.redhat.com/articles/2023/07/24/how-integrate-spring-boot-3-spring-security-and-keycloak
[04] https://apisix.apache.org/docs/apisix/plugins/openid-connect/#token-introspection
[05] https://documentation.cloud-iam.com/resources/keycloak-authentications/passwordless-webauthn.html
